components/brave_news/browser/resources/shared/configurationCache.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import {
+  BraveNewsControllerRemote,
+  Configuration,
+  ConfigurationListenerInterface,
+  ConfigurationListenerReceiver,
+} from 'gen/brave/components/brave_news/common/brave_news.mojom.m'
+import getBraveNewsController from './api'
+
+import { CachingWrapper } from '$web-common/mojomCache'
+
+export class ConfigurationCachingWrapper
+  extends CachingWrapper<Configuration>
+  implements ConfigurationListenerInterface {
+  private receiver = new ConfigurationListenerReceiver(this)
+  private controller: BraveNewsControllerRemote
+
+  constructor() {
+    super({
+      isOptedIn: false,
+      showOnNTP: false
+    })
+
+    this.controller = getBraveNewsController()
+
+    // We can't set up  the mojo pipe in the test environment.
+    if (process.env.NODE_ENV !== 'test') {
```

Comment:
Yup, I think it's compiled in as a global variable, via a webpack plugin:
https://webpack.js.org/plugins/environment-plugin/

I vaguely remember that it does some static analysis too, so this might even be compiled out?

Reply:
Didn't know about it! That's nice 👍